### PR TITLE
feat(lambda): enable RuntimeManagementConfig 

### LINKED ISSUE #23890: lambda: enable RuntimeManagementConfig
### Describe the feature

Introducing AWS Lambda runtime management controls
https://aws.amazon.com/jp/blogs/compute/introducing-aws-lambda-runtime-management-controls/

https://docs.aws.amazon.com/lambda/latest/dg/runtimes-update.html

### Use Case

```ts
const fn = new lambda.Function(this, 'MyFunction', {
  runtime: lambda.Runtime.NODEJS_18_X,
  runtimeManagement: {
    arn: "arn:aws:lambda:ap-northeast-1::runtime:07a48df201798d627f2b950f03bb227aab4a655a1d019c3296406f95937e2525",
    mode: lambda.updateRuntimeOn.AUTO
  }
  handler: 'index.handler',
  code: lambda.Code.fromAsset(path.join(__dirname, 'lambda-handler')),
});
```

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

v2.62.0

### Environment details (OS name and version, etc.)

any

### PR DESCRIPTION
**Pull request by watany-dev**
Introducing AWS Lambda runtime management controls
https://aws.amazon.com/jp/blogs/compute/introducing-aws-lambda-runtime-management-controls/

This setting achieves the following set values.
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html#cfn-lambda-functionruntimemanagementconfig

I have not been able to test this CFn as it does not seem to be supported by cdk. It's only a design.

Closes #23890.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-lambda/README.md [hunk 1]
```diff
@@ -1042,3 +1042,19 @@ new lambda.Function(this, 'Function', {
...
+## Runtime updates
+
+Lambda runtime management controls help reduce the risk of impact to your workloads in the rare event of a runtime version incompatibility.
+For more information, see [Runtime management controls](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-update.html#runtime-management-controls)
+
+```ts
+new Function(stack, 'Lambda', {
+  runtimeManagement: {
+    mode: UpdateRuntimeOn.AUTO,
+  },
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-lambda/README.md:1055]:
```suggestion
  runtimeManagementMode: UpdateRuntimeOn.AUTO,
```

File: packages/@aws-cdk/aws-lambda/lib/runtime-management.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+   * This is only required if you're using the Manual runtime update mode.
+   *
+   * @default none
+   */
+  readonly arn?: string;
+  /**
+   * Specify the runtime update mode.
+   */
+  readonly mode: UpdateRuntimeOn;
+}
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-lambda/lib/runtime-management.ts:46]: Can we remove this interface, and just have `runtimeManagementVersionArn` and `runtimeManagementMode` as props?

File: packages/@aws-cdk/aws-lambda/lib/function.ts [hunk 3]
```diff
@@ -779,6 +786,19 @@ export class Function extends FunctionBase {
       throw new Error(`Ephemeral storage size must be between 512 and 10240 MB, received ${props.ephemeralStorageSize}.`);
     }
 
+    let rmc = undefined;
+    if (props.runtimeManagement) {
+      if (props.runtimeManagement.mode === UpdateRuntimeOn.MANUAL) {
+        rmc = {
+          runtimeVersionArn: props.runtimeManagement.arn,
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-lambda/lib/function.ts:793]: we need to throw an error if `runtimeManagement.arn` is not defined

File: packages/@aws-cdk/aws-lambda/test/runtime-management.test.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      },
+    });
+    // WHEN
+    Template.fromStack(stack).hasResourceProperties('AWS::Lambda::Function', {
+      RuntimeManagementConfig: {
+        UpdateRuntimeOn: 'Function update',
+      },
+    });
+  });
+  test('Runtime Management MANUAL', () => {
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-lambda/test/runtime-management.test.ts:44]: we need a test to verify that the `manual` case throws an error if the `arn` is unspecified

File: packages/@aws-cdk/aws-lambda/test/runtime-management.test.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      },
+    });
+    // WHEN
+    Template.fromStack(stack).hasResourceProperties('AWS::Lambda::Function', {
+      RuntimeManagementConfig: {
+        RuntimeVersionArn: 'arn:aws:lambda:ap-northeast-1::runtime:07a48df201798d627f2b950f03bb227aab4a655a1d019c3296406f95937e2525',
+        UpdateRuntimeOn: 'Manual',
+      },
+    });
+
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-lambda/test/runtime-management.test.ts:64]:
```suggestion
```

File: packages/@aws-cdk/aws-lambda/test/runtime-management.test.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  });
+
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-lambda/test/runtime-management.test.ts:66]:
```suggestion
```

File: packages/@aws-cdk/aws-lambda/lib/function.ts [hunk 7]
```diff
@@ -359,6 +360,12 @@ export interface FunctionOptions extends EventInvokeConfigOptions {
    * @default Architecture.X86_64
    */
   readonly architecture?: Architecture;
+
+  /**
+   * Sets the runtime management configuration for a function's version.
+   * @default Auto
+   */
+  readonly runtimeManagement?: RuntimeManagement;
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-lambda/lib/function.ts:368]:
```suggestion
  readonly runtimeManagementMode?: RuntimeManagementMode;
  
  readonly runtimeManagementVersionArn?: string;
```

**Review by comcalvi — CHANGES_REQUESTED** [review]
Thanks for the PR, this looks pretty good. I'd prefer to keep this interface as flat props, since in most cases only one will need to be specified (I'm figuring most customers will prefer `auto` for runtime management)

**Comment by watany-dev** [comment]
Thank you. I have applied the patches I could, but I will try again.
The ARn part is still being designed, but I feel it would be easier to set it up this way

runtimeManagementMode: UpdateRuntimeOn.MANUAL_ARN(<someArn>)

**Comment by comcalvi** [comment]
why `UpdateRuntimeOn`? It's not obvious from the name `runtimeManagementMode` that you would need to specify a `UpdateRuntimeOn`

**Comment by watany-dev** [comment]
That is a fair point. I was using the original parameter names as they are, but I will consider a proposal to revise the naming to match.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-function-runtimemanagementconfig.html#cfn- lambda-function-runtimemanagementconfig-updateruntimeon

File: packages/@aws-cdk/aws-lambda/lib/function.ts (hunk 7 above)
Comment (watany-dev) [reply packages/@aws-cdk/aws-lambda/lib/function.ts]: Implemented as specified in `.manual(arn)` of RuntimeManagementMode.

File: packages/@aws-cdk/aws-lambda/lib/runtime-management.ts (hunk 2 above)
Comment (watany-dev) [reply packages/@aws-cdk/aws-lambda/lib/runtime-management.ts]: I changed this to flat.

**Review by comcalvi — APPROVED** [review]
nice work! lgtm

**Pull request merged**